l1-contracts/contracts/transactionFilterer/GatewayTransactionFilterer.sol

```diff
@@ -82,18 +83,28 @@ contract GatewayTransactionFilterer is ITransactionFilterer, ReentrancyGuard, Ow
     ) external view returns (bool) {
         if (sender == L1_ASSET_ROUTER) {
             bytes4 l2TxSelector = bytes4(l2Calldata[:4]);
+
+            if (IL2AssetRouter.setAssetHandlerAddress.selector == l2TxSelector) {
+                (, bytes32 decodedAssetId, ) = abi.decode(l2Calldata[4:], (uint256, bytes32, address));
+                return _checkSTMAssetId(decodedAssetId);
+            }
+
             if (
-                (IAssetRouterBase.finalizeDeposit.selector != l2TxSelector) &&
-                (IL2Bridge.finalizeDeposit.selector != l2TxSelector)
+                IAssetRouterBase.finalizeDeposit.selector != l2TxSelector &&
+                IL2Bridge.finalizeDeposit.selector != l2TxSelector
```

Comment:
Does this allow the chains to be withdrawn btw? We should also allow the withdraw function selector

Reply:
Let's move it to a different scope, I would want to implement only together with testing for it

Reply:
https://linear.app/matterlabs/issue/EVM-750/support-chain-withdrawals-from-gateway